### PR TITLE
Support password autofill and textContextType

### PR DESCRIPTION
**Pull request by eschramm**
Closes https://github.com/CareEvolution/RKStudio-Participant/issues/1023 (and https://github.com/CareEvolution/ResearchKit/pull/9).

This PR looks to add the functionality added in [Eric DeLabar's PR on base ResearchKit](https://github.com/ResearchKit/ResearchKit/pull/1188) which incorporates some of the improvements in password management that arrived in iOS 12.

A new property on `ORK[1]TextAnswerFormat` named `allowPasswordAutoFill` is used to provide a switch for the autofill-avoidant behavior added in [Kevin's fix for secure text fields in surveys](https://github.com/CareEvolution/ResearchKit/pull/61).

This will allow us to selectively allow password autofill and management for myFHR which still uses the `ORK1RegistrationStep` and `ORK1LoginStep`. (Note that this does not attempt to convert the myFHR login/registration screens to the newer style supported by AoU and MDH at this time.)

While this feature was added both to RK1 and RK2, I did not attempt to update the minimum version for the ResearchKit (RK2) framework at this time, hence, the `if (@available` statements in RK2 here.

### Password Functionality

For the login/appropriate domain matching, it's messy if you like me have numerous saved passwords for the careevolution.com domain. I made a simple dummy test login in iOS's Password database (Settings > Passwords) using a unique domain for each app (mydatahelps.org = MDH, myfhr.careevolution.com = myFHR, platform.joinallofus.org = AoU). I was able to confirm that I only saw the right test login for each app (except myFHR which matches on the wildcard *.careevolution.com so it showed in all 3 apps).

- MyDataHelps (basically regression testing)
  - ensure text answers marked "Sensitive" (`secureTextEntry = true`) fields in surveys  (QuestionStep and FormItems) continue to NOT suggest/show passwords nor ask to save them at survey completion.
  - login shows saved passwords for appropriate domains:
     - `mydatahelps.org`
     - `caev.io`
     - `mdhorg.ce.dev`
     - `*.careevolution.com`
- MyFHR
  - login should now appropriately suggest saved passwords if available
  - new account creation should be pretty friction-free if autofill passwords is enabled - when filling out the fields, it will suggest name, address, and a secure password (assuming Settings has this enabled - Settings > Passwords > AutoFill Passwords)
  - login shows passwords for appropriate domains:
    - `myfhr.careevolution.com`
    - `myfhr.careevolution.dev`
    - `caev.io`
- All Of Us (should not be otherwise affected)
  - login shows saved passwords for appropriate domains:
    - `platform.joinallofus.org`
    - `platformaou.ce.dev`
    - `caev.io`
    - `*.careevolution.com`

Since testing RK2 is not-trivial, I did a search and replace in RKStudio-Participant for much of the login screens from `ORK1` -> `ORK` and can confirm the behavior seen in RK1 applies to RK2. Here's a screenshot for giggles on what the myFHR screen would look like in RK2.

|Login|Registration|
|---|---|
|IMG_528F937D558A-1|IMG_094B2722BBB5-1|

### ContextType

This [survey JSON](https://gist.github.com/eschramm/da7cd4c544b23f29d18efde2439eac32#file-allfilecontexttypesformstep-json) has all the types as FormItems to see how suggestions look for the various types.

### CODE REVIEW

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h [hunk 1]
```diff
@@ -1345,6 +1345,28 @@ ORK1_CLASS_AVAILABLE
  */
 @property (nonatomic,getter=isSecureTextEntry) BOOL secureTextEntry;
 
+/**
+ If `secureTextEntry` is YES, determines if passwordAutofill is enabled.
+ 
+ By default, the value fo this property is NO.
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.h]: Typo: "fo"

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m [hunk 2]
```diff
@@ -2493,6 +2509,7 @@ - (ORK1AnswerFormat *)impliedAnswerFormat {
         _impliedAnswerFormat.spellCheckingType = UITextSpellCheckingTypeNo;
         _impliedAnswerFormat.autocapitalizationType = UITextAutocapitalizationTypeNone;
         _impliedAnswerFormat.autocorrectionType = UITextAutocorrectionTypeNo;
+        _impliedAnswerFormat.textContentType = UITextContentTypeEmailAddress;
```
Comment (mmertsock) [inline ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m]: Shouldn't this have the same `usernameField ? UITextContentTypeUsername : UITextContentTypeEmailAddress` logic that you're using below?

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 3]
```diff
@@ -1413,6 +1413,28 @@ ORK_CLASS_AVAILABLE
  */
 @property (nonatomic,getter=isSecureTextEntry) BOOL secureTextEntry;
 
+/**
+ If `secureTextEntry` is YES, determines if passwordAutofill is enabled.
+ 
+ By default, the value fo this property is NO.
```
Comment (mmertsock) [inline ResearchKit/Common/ORKAnswerFormat.h]: Typo: "fo"

**Review by mmertsock — APPROVED** [review]
LGTM, just some minor comments

**Review by mmertsock — APPROVED** [review]
~LGTM~ (ignore, posted to wrong PR)

File: ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m (hunk 2 above)
Comment (eschramm) [reply ORK1Kit/ORK1Kit/Common/ORK1AnswerFormat.m]: Good point, I think we DO want that - added

**Review by mmertsock — APPROVED** [review]
Latest changes LGTM 🚢

**Pull request merged**